gossipd/gossip_store.c

```diff
@@ -116,14 +116,44 @@ static bool append_msg(int fd, const u8 *msg, u32 timestamp,
 	return true;
 }
 
+#ifdef COMPAT_V082
+/* The upgrade from version 7 is trivial */
+static bool can_upgrade(u8 oldversion)
+{
+	return oldversion == 7;
+}
+
+static bool upgrade_field(u8 oldversion, u8 **msg)
+{
+	assert(can_upgrade(oldversion));
+
+	/* We only need to upgrade this */
+	if (fromwire_peektype(*msg) == WIRE_GOSSIPD_LOCAL_ADD_CHANNEL) {
+		/* Append two 0 bytes, for (empty) feature bits */
+		tal_resizez(msg, tal_bytelen(*msg) + 2);
+	}
+	return true;
+}
```

Comment:
Patching these on the fly seems a bit strange to me:  couldn't we have optional fields at the end, maybe even make the switch to TLV encode the `local_add_channel` instead? That'd allow us to add/remove optional fields without bumping the `gossip_store` version later. It'd be better future proofing imho.

Reply:
Well, these messages are unusual: usually we don't need to worry about compatibility at all. I only did this because it was easy: I don't want all the callers to have to worry about missing fields.

Reply:
Oh, BTW, these are patched during our "offline compaction" run which always runs at startup and rewrites the files.

So they're not converted every time!